eng/targetingpacks.targets

```diff
@@ -0,0 +1,106 @@
+<Project>
+  <PropertyGroup Condition="'$(DisableImplicitFrameworkReferences)' != 'true' and
+                            '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
+                            $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '$(NETCoreAppCurrentVersion)'))">
+    <_UseLocalTargetingRuntimePack>true</_UseLocalTargetingRuntimePack>
+    <EnableTargetingPackDownload>false</EnableTargetingPackDownload>
+  </PropertyGroup>
+
+  <!-- .NETCoreApp 2.x DisableImplicitAssemblyReferences support. -->
+  <Choose>
+    <When Condition="'$(DisableImplicitAssemblyReferences)' == 'true' and
+                     '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
+                     $([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '3.0'))">
+    <PropertyGroup>
+      <MicrosoftNetCoreAppRefPackRefDir>$(PkgMicrosoft_NETCore_App)\ref\$(_ShortFrameworkIdentifier)$(_ShortFrameworkVersion)\</MicrosoftNetCoreAppRefPackRefDir>
+    </PropertyGroup>
+    <ItemGroup>
+      <PackageReference Update="Microsoft.NETCore.App"
+                        ExcludeAssets="all"
+                        GeneratePathProperty="true" />
+    </ItemGroup>
+    </When>
+  </Choose>
+
+  <!-- Simple name references will be resolved from the targeting pack folders and should never be copied to the output. -->
+  <ItemGroup>
+    <Reference Update="@(Reference)">
+      <Private Condition="'%(Reference.Extension)' != '.dll'">false</Private>
+    </Reference>
+  </ItemGroup>
+
+  <!-- .NETCoreApp > 2.x DisableImplicitAssemblyReferences support. -->
+  <Target Name="RemoveFrameworkReferences"
+          BeforeTargets="_HandlePackageFileConflicts"
+          AfterTargets="ResolveTargetingPackAssets"
+          Condition="'$(DisableImplicitAssemblyReferences)' == 'true' and
+                     '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
+                     $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '3.0'))">
+    <PropertyGroup>
+      <MicrosoftNetCoreAppRefPackRefDir>%(ResolvedFrameworkReference.TargetingPackPath)\ref\$(_ShortFrameworkIdentifier)$(_ShortFrameworkVersion)\</MicrosoftNetCoreAppRefPackRefDir>
+    </PropertyGroup>
+    <ItemGroup>
+      <Reference Remove="@(Reference)"
+                 Condition="'%(Reference.FrameworkReferenceName)' == '$(SharedFrameworkName)'" />
+    </ItemGroup>
+  </Target>
+
+  <!-- Add the resolved targeting pack to the assembly search path. -->
+  <Target Name="UseTargetingPackForAssemblySearchPaths"
+          BeforeTargets="ResolveAssemblyReferences;
+                         DesignTimeResolveAssemblyReferences"
+          Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'">
+    <PropertyGroup>
+      <AssemblySearchPaths>$(AssemblySearchPaths);$(MicrosoftNetCoreAppRefPackRefDir.TrimEnd('/\'))</AssemblySearchPaths>
```

Comment:
We do we need this is we're already updating the ResolvedTargetingPack and RuntimePack?

Reply:
The AssemblySearchPath isn't set by the targetingpack and hence we need to set it manually. As we support both the targetingpack (>3.x) and the packagereference model (2.x) there is no common way when to set it so I sequenced it before it is used in `ResolveAssemblyReferences` and `DesignTimeResolveAssemblyReferences`.